src/app/treasure-hunt/treasure-hunt-report/executive-summary/executive-summary.component.ts

```diff
@@ -83,22 +98,70 @@ export class ExecutiveSummaryComponent implements OnInit {
     this.dataTitles = dataTitles;
   }
 
-  getChartHeight(): number {
-    if (this.costSummaryChartContainer) {
-      return this.costSummaryChartContainer.nativeElement.clientHeight;
+  prepPieChartData() {
+    this.pieChartValues = new Array<number>();
+    this.pieChartLabels = new Array<string>();
+    for (let i = 0; i < this.teamData.length; i++) {
+      this.pieChartLabels.push(this.teamData[i].team + ' $' + (Math.round(this.teamData[i].costSavings)));
+      this.pieChartValues.push(this.teamData[i].costSavings);
+    }
+  }
+
+  prepTeamData() {
+    let teams: Array<string> = new Array<string>();
+    this.teamData = new Array<any>();
+    for (let i = 0; i < this.opportunityCardsData.length; i++) {
+      if (!_.includes(teams, this.opportunityCardsData[i].teamName)) {
+        teams.push(this.opportunityCardsData[i].teamName);
+        let data = {
+          team: this.opportunityCardsData[i].teamName,
+          costSavings: this.opportunityCardsData[i].annualCostSavings,
+          implementationCost: this.opportunityCardsData[i].implementationCost,
+          payback: this.opportunityCardsData[i].paybackPeriod
+        };
+        this.teamData.push(data);
+      } else {
+        let index = _.indexOf(teams, this.opportunityCardsData[i].teamName);
+        this.teamData[index].costSavings += this.opportunityCardsData[index].annualCostSavings;
+        this.teamData[index].implementationCost += this.opportunityCardsData[index].implementationCost;
+        this.teamData[index].payback += this.opportunityCardsData[index].paybackPeriod;
+      }
+    }
+  }
+
+  getChartHeight(chart: string): number {
```

Comment:
For chart width/height values. Instead of binding a function to an input that causes the  "after view checked errors". Use variables that are set and passed into graph, set the height/width on ngAfterViewInit and then use HostListener to change them when the window changes sizes

@HostListener('window:resize', ['$event']){
 // call function that sets graph sizes
}

then use the ChangeDetectorRef to detect changes. Will eliminate errors. Ask me if you have questions.